### ISSUE TITLE
Are versions 11.6+ leaking memory?

### ISSUE THREAD

**Issue by mscreenie** [nwjs/nw.js#3374]
I have two instances of identical apps running, one in 11.6 and other 12.1. The app makes a ajax call every 10 seconds, its purpose, to keep sessions alive. 

Works fine with 11.6, We left it running on our test machine and after 3 days 11.6 had three processes using a total of 40-50 MB and stable.

Today I begun testing 12.1 and saw after 2 hours three processes using a total of 300 MB and increasing.

I know there are significant differences between the two Node => io.js and an updated Chromium. I'm wondering where the problem is, I've looked at my code and checked if there are any possible problems for GC but seems OK and both instances are using identical code. But I haven't ruled out my code being at fault, its always a real possibility. 

The problem is going to be interesting to debug, ill see if taking heap snapshots will help me identify the problem. If I was to guess I would say io.js is at fault. Chrome to leak memory on a stable release is rear, plus there have been some fixes for mem leaks on recent io.js releases. 

Still I'm a little worried now and stuck between 11.6 (which works fine except for an SSL bug) and 12.1. Hoping that updates to latest io.js may remedy the problem.
##

**Comment by KyleRoss** [comment]
I've noticed the same issue since moving to 12.1 myself. It seems to be consuming memory at incredible rates. I have a larger application written for NW and expect memory usage to be higher, but without doing anything in the application, the memory usage keeps rising at a consistent rate where I've peaked at 450MB before. I've done tons of profiling to reduce memory leaks in my application, although my heap sits at a consistent 22MB and the NW process just keeps rising. I often thought it could have been one of the modules I am using (ex. mongoose) that was causing it, although the heap would be rising along with it.
